diff --git a/dist/esm/components/elements/ssr-layout-effect.js b/dist/esm/components/elements/ssr-layout-effect.js
index 18703b4465ef31d04e1e890926fbbee5d71eeb3a..796df60c1433d16d5f61fead604e1561aff8fa7b 100644
--- a/dist/esm/components/elements/ssr-layout-effect.js
+++ b/dist/esm/components/elements/ssr-layout-effect.js
@@ -2,12 +2,8 @@
 "use client";
 
 // src/components/elements/ssr-layout-effect.tsx
-import { useLayoutEffect } from "react";
 import { jsx } from "react/jsx-runtime";
 function SsrScript(props) {
-  useLayoutEffect(() => {
-    (0, eval)(props.script);
-  }, []);
   return /* @__PURE__ */ jsx(
     "script",
     {
diff --git a/dist/esm/components/user-button.js b/dist/esm/components/user-button.js
index 1694184dda4b0292941625c64c7adcd15065fd8c..01a7452add7e19f4c3ae706ea10a7d15c7c269a1 100644
--- a/dist/esm/components/user-button.js
+++ b/dist/esm/components/user-button.js
@@ -91,6 +91,7 @@ function UserButtonInnerInner(props) {
   ] });
 }
 export {
-  UserButton
+  UserButton,
+  UserAvatar
 };
 //# sourceMappingURL=user-button.js.map
\ No newline at end of file
diff --git a/dist/esm/index.js b/dist/esm/index.js
index 844c365b227253fffac2f300516cc1fad2a47321..dd6ea84c5fada4448cc2de53695fe3144a07e32a 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -10,7 +10,7 @@ import { EmailVerification } from "./components-page/email-verification";
 import { PasswordReset } from "./components-page/password-reset";
 import { ForgotPassword } from "./components-page/forgot-password";
 import { MessageCard } from "./components/message-cards/message-card";
-import { UserButton } from "./components/user-button";
+import { UserButton, UserAvatar } from "./components/user-button";
 import { AccountSettings } from "./components-page/account-settings";
 import { AuthPage } from "./components-page/auth-page";
 import { CredentialSignIn } from "./components/credential-sign-in";
@@ -39,6 +39,7 @@ export {
   default3 as StackHandler,
   default2 as StackProvider,
   StackTheme,
+  UserAvatar,
   UserButton,
   useStackApp,
   useUser
