diff --git a/dist/esm/lib/cookie.js b/dist/esm/lib/cookie.js
index 61588854f6906d19c2bb7366ebfeeffbebd274ae..3f2d4d7e1a79055c12123806ad79d375b2aaa372 100644
--- a/dist/esm/lib/cookie.js
+++ b/dist/esm/lib/cookie.js
@@ -172,9 +172,6 @@ function setCookie(name, value, options = {}) {
     });
   } catch (e) {
     if (isRscCookieUnavailableError(e)) {
-      if (window.location.protocol !== "https:" && isProd) {
-        throw new Error("Attempted to set a secure cookie, but this build was compiled as a production build, but the current page is not served over HTTPS. This is a security risk and is not allowed in production.");
-      }
       Cookies.set(name, value, {
         secure: isProd,
         expires: options.maxAge === void 0 ? void 0 : new Date(Date.now() + options.maxAge * 1e3)
diff --git a/dist/lib/cookie.js b/dist/lib/cookie.js
index 3afc5db01c879d117614ba158238308ffeefd5c1..f8a10509115071fafd7c7282ed430f30efede2c0 100644
--- a/dist/lib/cookie.js
+++ b/dist/lib/cookie.js
@@ -211,9 +211,6 @@ function setCookie(name, value, options = {}) {
     });
   } catch (e) {
     if (isRscCookieUnavailableError(e)) {
-      if (window.location.protocol !== "https:" && isProd) {
-        throw new Error("Attempted to set a secure cookie, but this build was compiled as a production build, but the current page is not served over HTTPS. This is a security risk and is not allowed in production.");
-      }
       import_js_cookie.default.set(name, value, {
         secure: isProd,
         expires: options.maxAge === void 0 ? void 0 : new Date(Date.now() + options.maxAge * 1e3)
