diff --git a/dist/components/elements/ssr-layout-effect.js b/dist/components/elements/ssr-layout-effect.js
index ebaa18cdf3974bb17edd4ccb94dd6f22393441b3..c4ea30996a97f005cb43d2a790b2f771f60cb53f 100644
--- a/dist/components/elements/ssr-layout-effect.js
+++ b/dist/components/elements/ssr-layout-effect.js
@@ -25,12 +25,8 @@ __export(ssr_layout_effect_exports, {
   SsrScript: () => SsrScript
 });
 module.exports = __toCommonJS(ssr_layout_effect_exports);
-var import_react = require("react");
 var import_jsx_runtime = require("react/jsx-runtime");
 function SsrScript(props) {
-  (0, import_react.useLayoutEffect)(() => {
-    (0, eval)(props.script);
-  }, []);
   return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
     "script",
     {
diff --git a/dist/esm/components/elements/ssr-layout-effect.js b/dist/esm/components/elements/ssr-layout-effect.js
index 18703b4465ef31d04e1e890926fbbee5d71eeb3a..796df60c1433d16d5f61fead604e1561aff8fa7b 100644
--- a/dist/esm/components/elements/ssr-layout-effect.js
+++ b/dist/esm/components/elements/ssr-layout-effect.js
@@ -2,12 +2,8 @@
 "use client";
 
 // src/components/elements/ssr-layout-effect.tsx
-import { useLayoutEffect } from "react";
 import { jsx } from "react/jsx-runtime";
 function SsrScript(props) {
-  useLayoutEffect(() => {
-    (0, eval)(props.script);
-  }, []);
   return /* @__PURE__ */ jsx(
     "script",
     {
