diff --git a/dist/esm/components/elements/ssr-layout-effect.js b/dist/esm/components/elements/ssr-layout-effect.js
index 18703b4465ef31d04e1e890926fbbee5d71eeb3a..796df60c1433d16d5f61fead604e1561aff8fa7b 100644
--- a/dist/esm/components/elements/ssr-layout-effect.js
+++ b/dist/esm/components/elements/ssr-layout-effect.js
@@ -2,12 +2,8 @@
 "use client";
 
 // src/components/elements/ssr-layout-effect.tsx
-import { useLayoutEffect } from "react";
 import { jsx } from "react/jsx-runtime";
 function SsrScript(props) {
-  useLayoutEffect(() => {
-    (0, eval)(props.script);
-  }, []);
   return /* @__PURE__ */ jsx(
     "script",
     {
