diff --git a/esm/vs/base/common/worker/simpleWorker.js b/esm/vs/base/common/worker/simpleWorker.js
index 85005f72fa8f356259711355424370f4b52acb8f..448465d024d6126c04878695c78f684bd703944c 100644
--- a/esm/vs/base/common/worker/simpleWorker.js
+++ b/esm/vs/base/common/worker/simpleWorker.js
@@ -425,7 +425,11 @@ export class SimpleWorkerServer {
             globalThis.require.config(loaderConfig);
         }
         if (isESM) {
-            const url = FileAccess.asBrowserUri(`${moduleId}.js`).toString(true);
+        const url = FileAccess.asBrowserUri(`${moduleId}.js`).toString(true);
+            if (process.env.NODE_ENV === "development") {
+                throw new Error(`No RequestHandler!`);
+                return;
+            }
             return import(`${url}`).then((module) => {
                 this._requestHandler = module.create(this);
                 if (!this._requestHandler) {
diff --git a/esm/vs/editor/common/services/editorSimpleWorker.js b/esm/vs/editor/common/services/editorSimpleWorker.js
index 91df4d4a6cda0a903a6e205b151e2a54e0d2098b..13d4e8240c88251d182dacce83c87c967260db66 100644
--- a/esm/vs/editor/common/services/editorSimpleWorker.js
+++ b/esm/vs/editor/common/services/editorSimpleWorker.js
@@ -320,6 +320,10 @@ export class EditorSimpleWorker extends BaseEditorSimpleWorker {
             }
             else {
                 const url = FileAccess.asBrowserUri(`${moduleId}.js`).toString(true);
+                if (process.env.NODE_ENV === "development") {
+                    reject();
+                    return;
+                }
                 import(`${url}`).then(onModuleCallback).catch(reject);
             }
         });
